### PR TITLE
Add get safe creation transaction

### LINKED ISSUE #241: Domain get safe creation transaction
# Description
Get transactions history endpoint add at the end the creation transaction for the safe because `all-transactions` endpoint on tx-service doesn't include this transaction. 
See the current implementation of the [client-gateway](https://github.com/safe-global/safe-client-gateway/blob/main/src/routes/transactions/handlers/history.rs#L62) 

# Proposal
Add safe domain method to return this transaction and add it at the end of the transactions history.

### PR DESCRIPTION
**Pull request by moisses89**
Closes #241 
- Add domain get safe creation transaction

# Additional notes
Couldn't keep alphabetic sort in [domain.module](https://github.com/5afe/safe-client-gateway-nest/compare/add_safe_creation_transaction?expand=1#diff-7d821304f2e0af7a603b971363f1bb70cb50afde8e2254719eb910536255ed27R71) because `creationTransactionValidator` contains a field that needs `dataDecodeValidator`.

### CODE REVIEW

File: src/domain/safe/entities/creation-transaction.entity.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { DataDecoded } from '../../data-decoder/entities/data-decoded.entity';
+
+export interface CreationTransaction {
+  created: Date;
+  creator: string;
+  transactionHash: string;
+  factoryAddress: string | null;
```
Comment (hectorgomezv) [inline src/domain/safe/entities/creation-transaction.entity.ts:7]: Is `factoryAddress` nullable? In Swagger it seems it's required (also `masterCopy` and `setupData`, but maybe that's a mistake), and [in the code](https://github.com/safe-global/safe-transaction-service/blob/d1a8aaa9e30902d1028998d109bff5d44be5218c/safe_transaction_service/history/services/safe_service.py#L40) it seems it's not optional 🤔

Comment (moisses89) [reply src/domain/safe/entities/creation-transaction.entity.ts:7]: You are right, I take from client-gateway https://github.com/safe-global/safe-client-gateway/blob/main/src/common/models/backend/transactions.rs#L97

Comment (moisses89) [reply src/domain/safe/entities/creation-transaction.entity.ts]: Done here https://github.com/5afe/safe-client-gateway-nest/pull/243/commits/9e42be09bd60b88af2d1e6a28398bc06f2360b00

**Review by hectorgomezv — APPROVED** [review]
🚀

**Pull request merged**